pkg/multiboot/multiboot_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multiboot
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+)
+
+func TestParseMemoryRanges(t *testing.T) {
+	root, err := ioutil.TempDir("", "memmap")
+	if err != nil {
+		t.Fatalf("Cannot create test dir: %v", err)
+	}
+	defer os.RemoveAll(root)
+	var want memoryRanges
+	for i := 0; i < 4; i++ {
+		p := path.Join(root, fmt.Sprint(i))
+		if err := os.Mkdir(p, 0755); err != nil {
+			t.Fatalf("Cannot create test dir: %v", err)
+		}
+
+		var typ rangeType
+		switch i {
+		case 0:
+			typ = rangeRAM
+		case 1:
+			typ = rangeNVACPI
+		case 2:
+			typ = rangeACPI
+		case 3:
+			typ = rangeReserved
+		}
+
+		start := uintptr(i * 100)
+		end := uintptr(i*100 + 50)
+		if err := ioutil.WriteFile(path.Join(p, "start"), []byte(fmt.Sprintf("0x%x\n", start)), 0655); err != nil {
```

Comment:
This also works for hex in Go:

```suggestion
		if err := ioutil.WriteFile(path.Join(p, "start"), []byte(fmt.Sprintf("%#x\n", start)), 0655); err != nil {
```

Reply:
nice, fixed